sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -485,27 +485,27 @@ enum EvictionSignal {
     PidAvailable,
 }
 
-impl TryFrom<&str> for KubernetesEvictionHardKey {
+impl TryFrom<&str> for KubernetesEvictionKey {
     type Error = error::Error;
 
     fn try_from(input: &str) -> Result<Self, Self::Error> {
         serde_plain::from_str::<EvictionSignal>(input).context(error::InvalidPlainValueSnafu {
             field: "Eviction Hard key",
         })?;
-        Ok(KubernetesEvictionHardKey {
+        Ok(KubernetesEvictionKey {
             inner: input.to_string(),
         })
     }
 }
-string_impls_for!(KubernetesEvictionHardKey, "KubernetesEvictionHardKey");
+string_impls_for!(KubernetesEvictionKey, "KubernetesEvictionKey");
```

Comment:
Can you convert these to use the new derive macro since they are being touched? You will no longer need a separate Enum and Struct, you can just have the enum.

See #2597, `sources/models/src/modeled_types/ecs.rs`